plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt

```diff
@@ -1017,8 +780,146 @@ class GesturePluginTest {
+    every { mapDelegateProvider.mapTransformDelegate } returns mapTransformDelegate
+    every { mapDelegateProvider.mapPluginProviderDelegate } returns mapPluginProviderDelegate
+    every { mapPluginProviderDelegate.getPlugin<CameraAnimationsPlugin>(Plugin.MAPBOX_CAMERA_PLUGIN_ID) } returns cameraAnimationsPlugin
+
+    presenter = GesturesPluginImpl(context, attrs, mockk(relaxed = true))
+
+    presenter.bind(context, gesturesManager, attrs, 1f)
+    presenter.onDelegateProvider(mapDelegateProvider)
+    presenter.initialize()
+  }
+
+  @Test
+  fun testFling() {
+    every { mapCameraManagerDelegate.cameraState } returns CameraState(
+      Point.fromLngLat(0.0, 0.0),
+      EdgeInsets(0.0, 0.0, 0.0, 0.0),
+      0.0,
+      0.0,
+      targetPitch
+    )
+    every {
+      mapCameraManagerDelegate.getDragCameraOptions(
+        any(),
+        any()
+      )
+    } returns CameraOptions.Builder().build()
+    presenter.updateSettings { scrollMode = targetScrollMode }
+    val motionEvent = mockk<MotionEvent>()
+    every { motionEvent.x } returns 0.0f
+    every { motionEvent.y } returns 0.0f
+    val result = presenter.handleFlingEvent(motionEvent, mockk(), velocity.first, velocity.second)
+    verify {
+      mapCameraManagerDelegate.getDragCameraOptions(
+        ScreenCoordinate(0.0, 0.0),
+        expectedCoordinate
+      )
+    }
+    verify { cameraAnimationsPlugin.easeTo(any(), any()) }
+    assert(result)
+  }
 
   private companion object {
+    @JvmStatic
+    @ParameterizedRobolectricTestRunner.Parameters(name = "Fling at pitch {0} with scroll mode {1} and velocity {2} should end at screen coordinate {3}")
+    fun data() = listOf(
+      arrayOf(
+        0.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(FLING_VELOCITY, FLING_VELOCITY),
+        ScreenCoordinate(FLING_DISPLACEMENT, FLING_DISPLACEMENT)
+      ),
+      arrayOf(
+        55.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(FLING_VELOCITY, FLING_VELOCITY),
+        ScreenCoordinate(645.1612903225806, 645.1612903225806)
+      ),
+      arrayOf(
+        80.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(FLING_VELOCITY, FLING_VELOCITY),
+        ScreenCoordinate(67.93869850933991, 67.93869850933991)
+      ),
+      arrayOf(
+        0.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(800f, 750f),
+        ScreenCoordinate(80.0, 75.0)
+      ),
+      arrayOf(
+        55.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(800f, 750f),
+        ScreenCoordinate(51.61290322580645, 48.38709677419355)
+      ),
+      arrayOf(
+        80.0,
+        ScrollMode.HORIZONTAL_AND_VERTICAL,
+        Pair(800f, 750f),
+        ScreenCoordinate(5.435095880747194, 5.095402388200494)
+      ),
+      arrayOf(
+        0.0,
+        ScrollMode.VERTICAL,
+        Pair(FLING_VELOCITY, FLING_VELOCITY),
+        ScreenCoordinate(0.0, FLING_DISPLACEMENT)
+      ),
+      arrayOf(
+        0.0,
```

Comment:
minimum value `0.0`? what is the maximum potential value?

Reply:
This is the pitch value, there's a constant value defined in GesturesConstants.kt:
```
internal const val MINIMUM_PITCH = 0.0
internal const val MAXIMUM_PITCH = 85.0
```

Reply:
but we have 80 as max value
maybe not super important to test it. up to you.
just want double check it :slightly_smiling_face:

Reply:
In the tests we covered no pitch(0) which is most frequent use case, low pitch(<60) and high pitch(between 60 and 85), which matches the internal logic where we applied different pitch factors for, so I think it should be good enough to cover these pitch bands 🙂